c7n/resources/ecs.py

```diff
@@ -594,25 +594,56 @@ class DeleteTaskDefinition(BaseAction):
     The definition will be marked as InActive. Currently running
     services and task can still reference, new services & tasks
     can't.
+    
+    force is False by default. When given as True, the task definition will 
+    be permanently deleted.
+    
+    .. code-block:: yaml
+
+       policies:
+         - name: deregister-task-definition
+           resource: ecs-task-definition
+           filters:
+             - family: test-task-def
+           actions:
+             - type: delete
+
+         - name: delete-task-definition
+           resource: ecs-task-definition
+           filters:
+             - family: test-task-def
+           actions:
+             - type: delete
+               force: True
     """
 
-    schema = type_schema('delete')
-    permissions = ('ecs:DeregisterTaskDefinition',)
+    schema = type_schema('delete', force={'type': 'boolean'})
+    permissions = ('ecs:DeregisterTaskDefinition','ecs:DeleteTaskDefinitions',)
 
     def process(self, resources):
         client = local_session(self.manager.session_factory).client('ecs')
         retry = get_retry(('Throttling',))
+        force = self.data.get('force', False)
 
         for r in resources:
+            if r['status'] == 'INACTIVE':
```

Comment:
I don't think this is needed because by default only ACTIVE task definitions are listed (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs/client/list_task_definitions.html)

Reply:
oh good call, eh... just noticed this comment after merging. i was thinking folks might use this action to delete tasks that were previously deregistered, but i guess that would need some more work to actually fetch the inactive ones to begin with :thinking: . that could be a separate change i guess, if it was worthwhile. thanks :+1:

Reply:
Yup, no worries. I guess we could figure out a way to expose the INACTIVE ones through a filter maybe?? Pretty sure it's gonna be a popular use case :)